### PR TITLE
Refresh UI

### PR DESCRIPTION
**Pull request by preston-willis**

### CODE REVIEW

File: src/Components/Layout.tsx [hunk 1]
```diff
@@ -58,12 +63,37 @@ export function Layout() {
     }
   }
 
+  function toMilliseconds() {
```
Comment (bennbollay) [inline src/Components/Layout.tsx]: Use the ms library

File: src/Components/Layout.tsx [hunk 2]
```diff
@@ -58,12 +63,37 @@ export function Layout() {
...
+      milliseconds = 1000*60*60*24-1
+    }
+    return milliseconds
+  }
+
+  function timer() {
+    if (timerRunning == true) {
+      return <Timer initialTime={toMilliseconds()} direction="backward">{({timerState}) => (<React.Fragment>{"Refresh in "}<Timer.Hours/>{":"}<Timer.Minutes/>{":"}<Timer.Seconds/></React.Fragment>)}</Timer>
+    } else {
+      return <div>Refesh</div>
```
Comment (bennbollay) [inline src/Components/Layout.tsx]: Spelling

File: src/Components/Layout.tsx [hunk 3]
```diff
@@ -58,12 +63,37 @@ export function Layout() {
...
+  }
+
     return (
       <div className='layout'>
         <Tabs onSelect={index => toggleDropdown(index)}>
           <TabList className='header'>
             <Tab>
-              <Button color='primary' variant='contained' className='header-button'>Menu</Button>
+              <Button color='primary' variant='contained' className='header-button' >
+              {timer()}
```
Comment (bennbollay) [inline src/Components/Layout.tsx]: Can you reduce the number of functions involved and include the above code directly? Or create an additional component?

File: src/Components/Menu.tsx [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
+  refreshInterval: number;
+  setRefreshInterval(x): void;
+  setTimerRunning(x): void;
+  timerRunning: boolean;
+  setMenuError(x): void;
+  menuError: boolean;
 }
 
-export function Menu(Props: Inputs) {
+export function MenuView(Props: Inputs) {
```
Comment (bennbollay) [inline src/Components/Menu.tsx]: Parameter should be lowercase first letter.

**Review by bennbollay — APPROVED** [review]
Can you perhaps reduce the number of parameters you pass by having some of them combined into one object?

File: src/Components/Menu.tsx (hunk 4 above)
Comment (preston-willis) [reply src/Components/Menu.tsx]: Are you talking about "Props"?

Comment (bennbollay) [reply src/Components/Menu.tsx]: Yes.

**Pull request merged**